### ISSUE TITLE
Run with `set -e`

### ISSUE THREAD

**Issue by giorgiosironi** [asm89/smoke.sh#11]
When I run with the bash `-e` option (stop at first command with a return code different from `0`), the tests terminate at the first check because of this increment:
```
+ _smoke_success '2xx Response code'
+ REASON='2xx Response code'
+ _smoke_print_success '2xx Response code'
+ TEXT='2xx Response code'
+ echo '    [ OK ] 2xx Response code'
    [ OK ] 2xx Response code
+ ((  SMOKE_TESTS_RUN++  ))
```
I think this could be tweaked because incrementing 0 returns a `1` code which stops the script.

**Comment by asm89** [comment]
Good catch. I'm afraid `smoke.sh` isn't really `set -e` proof regardless as it relies on checking the exit code manually in various places as the moment.
